### PR TITLE
Filter out default mounts that are being overriden by the user.

### PR DESCRIPTION
**Pull request by mrunalp**
This allows one to supply own /dev/shm or /dev/mqueue to multiple containers so they can communicate over POSIX IPC.

```
[root@localhost ~]# mount -t tmpfs -o size=10m tmpfs /volumes/dev/shm
[root@localhost ~]# /root/gosrc/src/github.com/docker/docker/bundles/1.5.0-dev/dynbinary/docker-1.5.0-dev run --rm -it -v /volumes/dev/shm/:/dev/shm fedora:21 bash                                                                       
bash-4.3# mount
/dev/mapper/docker-252:3-265558-20c772825b1933b57fca02bea4c249922fbe926c18f04ffd25aed18059b07e9d on / type ext4 (rw,relatime,seclabel,discard,stripe=16,data=ordered)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev type tmpfs (rw,nosuid,seclabel,mode=755)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,seclabel,gid=5,mode=620,ptmxmode=666)
mqueue on /dev/mqueue type mqueue (rw,nosuid,nodev,noexec,relatime,seclabel)
sysfs on /sys type sysfs (ro,nosuid,nodev,noexec,relatime,seclabel)
tmpfs on /dev/shm type tmpfs (rw,relatime,seclabel,size=20480k)
/dev/vda3 on /etc/resolv.conf type ext4 (rw,relatime,seclabel,data=ordered)
/dev/vda3 on /etc/hostname type ext4 (rw,relatime,seclabel,data=ordered)
/dev/vda3 on /etc/hosts type ext4 (rw,relatime,seclabel,data=ordered)
devpts on /dev/console type devpts (rw,nosuid,noexec,relatime,seclabel,gid=5,mode=620,ptmxmode=000)
proc on /proc/sys type proc (ro,nosuid,nodev,noexec,relatime)
proc on /proc/sysrq-trigger type proc (ro,nosuid,nodev,noexec,relatime)
proc on /proc/irq type proc (ro,nosuid,nodev,noexec,relatime)
proc on /proc/bus type proc (ro,nosuid,nodev,noexec,relatime)
tmpfs on /proc/kcore type tmpfs (rw,nosuid,seclabel,mode=755)

Note that there is no default /dev/shm mount in the output above.
```

Signed-off-by: Mrunal Patel mrunalp@gmail.com

@crosbymichael @LK4D4 ping

### CODE REVIEW

**Comment by LK4D4** [comment]
Omg, arrays sucks :)
@mrunalp Also, I'm working now on deleting `execdriver` totally. I'll include your fix there, WDYT?

**Comment by mrunalp** [comment]
@LK4D4 Thank you :) that works for me!

File: daemon/execdriver/native/create.go [hunk 1]
```diff
@@ -212,6 +212,20 @@ func (d *driver) setupRlimits(container *configs.Config, c *execdriver.Command)
 }
 
 func (d *driver) setupMounts(container *configs.Config, c *execdriver.Command) error {
+	userMounts := make(map[string]bool)
```
Comment (icecrime) [inline daemon/execdriver/native/create.go]: That's really a detail, but isn't the idiomatic way to implement a `set` in Go to have a `map[string]struct{}` and use `if _, ok := userMounts[m.Destination]; !ok { ... }`?

Comment (mrunalp) [inline daemon/execdriver/native/create.go]: @icecrime https://golang.org/doc/effective_go.html#maps describes using map[string]bool as a Set.

Comment (icecrime) [inline daemon/execdriver/native/create.go]: Oh alright :+1:

Comment (jlhawn) [inline daemon/execdriver/native/create.go]: @icecrime @mrunalp It's more of a personal/team/project preference. The `map[Type]bool` is a bit cleaner code-wise, but the `map[Type]struct{}` uses less memory. For this reason, the Docker project tends toward the `map[Type]struct{}` idiom instead.

Comment (jlhawn) [inline daemon/execdriver/native/create.go]: in a more perfect world, Go would have a builtin `set[Type]` type

Comment (mrunalp) [inline daemon/execdriver/native/create.go]: IMO, Go should have templates so users could create whatever datatypes they want ;)

**Comment by mrunalp** [comment]
@icecrime @jlhawn updated.

**Comment by crosbymichael** [comment]
LGTM

**Comment by jfrazelle** [comment]
LGTM

**Pull request merged**